memento-cassandra.root/usr/share/clearwater/cassandra-schemas/memento.sh

```diff
@@ -1,36 +1,42 @@
 #! /bin/bash
-. /usr/share/clearwater/cassandra_schema_utils.sh
 
 cassandra_hostname="127.0.0.1"
 
 . /etc/clearwater/config
+. /usr/share/clearwater/utils/check-root-permissions 1
+. /usr/share/clearwater/cassandra_schema_utils.sh
 
 quit_if_no_cassandra
 
+echo "Adding/updating Cassandra schemas..."
+
+# Wait for the cassandra cluster to come online
+count=0
+/usr/share/clearwater/bin/poll_cassandra.sh --no-grace-period > /dev/null 2>&1
+
+while [ $? -ne 0 ]; do
+  ((count++))
+  if [ $count -gt 120 ]; then
+    echo "Cassandra isn't responsive, unable to add/update schemas yet"
+    exit 1
+  fi
+
+  sleep 1
+  /usr/share/clearwater/bin/poll_cassandra.sh --no-grace-period > /dev/null 2>&1
+
+done
+
 CQLSH="/usr/share/clearwater/bin/run-in-signaling-namespace cqlsh"
 
-if [[ ! -e /var/lib/cassandra/data/memento ]] || \
+rc=0
+
+if [[ ! -e /var/lib/cassandra/data/cedar ]] || \
    [[ $cassandra_hostname != "127.0.0.1" ]];
 then
-  echo "Adding Cassandra schemas..."
-  count=0
-  /usr/share/clearwater/bin/poll_cassandra.sh --no-grace-period > /dev/null 2>&1
-
-  while [ $? -ne 0 ]; do
-    ((count++))
-    if [ $count -gt 120 ]; then
-      echo "Cassandra isn't responsive, unable to add schemas"
-      exit 1
-    fi
-
-    sleep 1
-    /usr/share/clearwater/bin/poll_cassandra.sh --no-grace-period > /dev/null 2>&1
-  done
-
-  # replication_str is set up by
-  # /usr/share/clearwater/cassandra-schemas/replication_string.sh
```

Comment:
(up to you)
I love comments and think we should keep this, but I'm happy for you to leave it deleted if you want.

Reply:
Well, it's no longer accurate. I think a corrected version isn't necessary though.

Reply:
ok got it!